### PR TITLE
NCR-14625 add Apply

### PR DESCRIPTION
**Pull request by solokirrik**

### CODE REVIEW

**Review by marksalpeter — CHANGES_REQUESTED** [review]
This looks like a useful addition to the lib and I appreciate the test coverage. This needs a `make readme` to update the readme file.

File: CODEOWNERS [hunk 1]
```diff
@@ -1 +1 @@
-* mark.salpeter@deliveryhero.com
\ No newline at end of file
+* @deliveryhero/adtech-tracking-billing-sdk
```
[the remark this answers is not in the document]
Comment (marksalpeter) [reply CODEOWNERS:1]: maybe dh_adtech_tracking@deliveryhero.com is a good email to use?

File: merge_example_test.go [hunk 2]
```diff
@@ -2,11 +2,12 @@ package pipeline_test
 
 import (
 	"fmt"
+	"testing"
 
 	"github.com/deliveryhero/pipeline/v2"
 )
 
-func ExampleMerge() {
+func TestExampleMerge(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (marksalpeter) [reply merge_example_test.go:10]: heres some context to learn more about why examples are in test files: https://go.dev/blog/examples

File: CODEOWNERS (hunk 1 above)
[the remark this answers is not in the document]
Comment (solokirrik) [reply CODEOWNERS]: Does not work. It should be a user. 
Reverted

**Review by marksalpeter — APPROVED** [review]
Looks good. Nice feature 👍

**Comment by marksalpeter** [comment]
Can you add an example for the new `Apply` function in the readme/godoc?

**Pull request merged**